[android][expokit] Release Android 2.11.1, 33.0.1 with expo-av@5.0.1

sjchmiela opened the pull request.

sjchmiela committed:
[repo] Ensure all workspace dependencies are symlinked

brentvatne committed:
[expo] Release 33.0.1

satuevsky committed:
[android] Fix AudioFocusNotAcquiredException

If mStaysActiveInBackground is true then don't throw the AudioFocusNotAcquiredException.

EvanBacon committed:
[storybook] deleted Loki (#4372)

* Delete .gitignore

* Delete chrome_laptop_APIs_Constants.png

* Delete chrome_laptop_Components_BlurView.png

* Delete chrome_laptop_Components_Linear_Gradient.png

* Update config.js

* Update package.json

EvanBacon committed:
[storybook] Fix component packages (#4373)

* Update package.json

* Update yarn.lock

* Update BlurViewScreen.js

* Update LinearGradientScreen.js

* Update BlurViewScreen.js

* Update LinearGradientScreen.js

* Update LinearGradientScreen.js

EvanBacon committed:
[storybook] Jest snapshots (#4379)

* added jest snapshots

* Removed loki command

* deleted loki

* Update babel.config.js

* Update config.js

* Update addons.js

* Update babel.config.js

* [WIP]

* Fixed storybook in workspace

* Update config.js

* Update package.json

* Update package.json

* Update package.json

* Update yarn.lock

* Update storyshots.test.js

* Create storyshots-test-js-image-ap-is-constants-1-snap.png

* Create storyshots-test-js-image-components-blur-view-1-snap.png

* Create storyshots-test-js-image-components-linear-gradient-1-snap.png

* Update storyshots.test.js.snap

* Added storybook test to CI

* Update config.yml

* Update storyshots-test-js-image-components-blur-view-1-snap.png

* Update storyshots.test.js

* Update package.json

* Added image params for CI

* Update storyshots.test.js.snap

* Update storyshots.test.js

* Update black-128x256.png

* Update example_image_1.jpg

* Update exponent-icon@2x.png

* Update exponent-icon@3x.png

* Update app.png

* Update loading.png

* Update qrcode_photo_wikipedia.jpg

* Update roadster.jpg

* Update package.json

* Update BlurViewScreen.js

* Update LinearGradientScreen.js

* Update package.json

* Update storyshots-test-js-image-components-blur-view-1-snap.png

* Update BlurViewScreen.js

* Update LinearGradientScreen.js

* Update AppText.js

* Update Code.js

* Update DocItem.js

* Create rem.js

* Update Section.js

* Update StyleList.js

* Update TextList.js

* Update UIExplorer.js

* Update package.json

* Update storyshots.test.js.snap

* Made style more generic

* Disabled visual regression testing in CI

tsapeta committed:
[ci] add shell_app_ios_build job to expo repo (#4355)

# Why

Shell apps could be built on the CI of this repo instead of the universe. 

# How

I've moved building iOS shell apps from the universe and also merged two jobs (for simulator and for real devices) into one job `shell_app_ios_build` that also uploads the tarball at the end.
I also placed iOS and Android jobs under one workflow `shell_app` that will be available only on the release branches and branches with `all-ci` substring (for testing) - I don't think we need to build shell apps on every commit as it happened so far (for iOS simulator), but I'm open to discussion 😉 

_bonus:_ Also fixed a bug in `expo_client_build` job that caused the tarball not to be uploaded to S3 😅 

# Test Plan

I've been testing this new `shell_app` workflow on my `@tsapeta/all-ci` branch, where I successfully built iOS shell app.

tsapeta committed:
[expotools] add update-turtle-sdk-version command (#4381)

# Why

Moving things from universe to the open repo.

# How

Moved `update-turtle-sdk-version` script from powertools in universe to expotools.

# Test Plan

Successfully updated Turtle SDK version to SDK33.

sjchmiela committed:
[sdk33] Backport 2ca24bffd to SDK33

sjchmiela committed:
[android] Bump Android versions to 2.11.1 (111)

sjchmiela committed:
[expo-av] Publish 5.0.1 with a fix for playing sounds when the app is in background

expo-av@5.0.1

sjchmiela committed:
[expokit] Update expokit with fixed expo-av@5.0.1

sjchmiela committed:
[expokit] Publish expokit@33.0.1 with expo-av@5.0.1

tsapeta committed:
[module-scripts] update jest-preset to include transforms from jest-expo

tsapeta committed:
[expo-camera] fix 'yarn build' returning TypeScript errors

sjchmiela committed:
[ci] Add FASTLANE_SKIP_UPDATE_CHECK env var to CI (#4384)

Up until now a non-latest version of Fastlane made the job fail (return code was 0).

tsapeta committed:
[contacts] fix 'yarn build' returning TypeScript errors

tsapeta committed:
[ios] fix react-native-screens on SDK32

Fixes #4371

tsapeta committed:
[changelog] cut off 33.0.0 version in changelog

ide committed:
[fastlane] Remove bg location automatically as part of the "release" lane

ide committed:
[ios] Update metadata for Fastlane

ide committed:
[jest] Include jest-preset.js in the published files

tsapeta committed:
[fastlane] restore original Info.plist after all lanes (#4404)

# Why

We modify Info.plist file during fastlane actions, but these changes are not cleared after that.

# How

Added saving the original Info.plist file before a lane and restoring it once lane finishes.

# Test Plan

Tested on a temporary lane that only configures for the app store submission. Lane succeeded and after that I didn't have any changes in Info.plist.

sjchmiela committed:
[tools] Clean build folders before building AARs

sjchmiela committed:
[expo] Make imports change message more prominent (#4389)

# Why

If the users don't make the `expo install` change, custom Android build system will strip out all the optional unimodules.

# How

Swapped order of tasks so that sufficient task follows the required one.

# Test Plan

The log is changed:

```
Running application on Moto G Play.
Following APIs have moved to separate packages and importing them from the 'expo' package is deprecated: Asset, Constants, Font, Icon.

1. Add correct versions of these packages to your project using:

   expo install @expo/vector-icons expo-asset expo-constants expo-font

   If "install" cannot be recognized as an expo command, update your expo-cli installation.

2. Change your imports so they don't use 'expo' package, but specific packages instead:

 - import { Asset } from 'expo' -> import { Asset } from 'expo-asset'
 - import { Constants } from 'expo' -> import Constants from 'expo-constants'
 - import { Font } from 'expo' -> import * as Font from 'expo-font'
 - import { Icon } from 'expo' -> import * as Icon from '@expo/vector-icons'
```

**Note: In the app I tested this on, I imported `WebBrowser` from `expo` and it did not end up in this message. 🤷‍♂**

ide committed:
[speech] Fix voiceIOS -> voice; update NCL

The "voiceIOS" option was renamed to "voice" -- this needs to be reflected in the TS types and also updated in NCL. Tested in NCL; the different voices work now.

ide committed:
Make deprecatedModule compile

Fix a string bug.

wkozyra95 committed:
[android] update jsc version to 241213.2.0

wkozyra95 committed:
[android] force jsc in standalone builds

sjchmiela committed:
[android] Fix AndroidManifest.xml by removing screenOrientation for LauncherActivity

esamelson committed:
[android] bump versions to 2.11.2 (112)

ide committed:
Update packages

expo-speech@5.0.1
expo-web-browser@5.0.1
expo-module-template@5.0.1
expo@33.0.2
expo-contacts@5.0.1
expo-image-picker@5.0.1

ide committed:
Update all packages

@unimodules/core@2.0.1
@unimodules/react-native-adapter@2.0.1
unimodules-barcode-scanner-interface@2.0.1
unimodules-camera-interface@2.0.1
unimodules-constants-interface@2.0.1
unimodules-face-detector-interface@2.0.1
unimodules-file-system-interface@2.0.1
unimodules-font-interface@2.0.1
unimodules-image-loader-interface@2.0.1
unimodules-permissions-interface@2.0.1
unimodules-sensors-interface@2.0.1
unimodules-task-manager-interface@2.0.1
expo-ads-admob@5.0.1
expo-ads-facebook@5.0.1
expo-analytics-amplitude@5.0.1
expo-analytics-segment@5.0.1
expo-app-auth@5.0.1
expo-app-loader-provider@5.0.1
expo-av@5.0.2
expo-background-fetch@5.0.1
expo-barcode-scanner@5.0.1
expo-blur@5.0.1
expo-brightness@5.0.1
expo-calendar@5.0.1
expo-camera@5.0.1
expo-constants@5.0.1
expo-contacts@5.0.2
expo-crypto@5.0.1
expo-document-picker@5.0.1
expo-facebook@5.0.1
expo-face-detector@5.0.1
expo-file-system@5.0.1
expo-font@5.0.1
expo-gl@5.0.1
expo-gl-cpp@5.0.1
expo-google-sign-in@5.0.1
expo-haptics@5.0.1
expo-image-manipulator@5.0.1
expo-image-picker@5.0.2
expo-intent-launcher@5.0.1
expo-keep-awake@5.0.1
expo-linear-gradient@5.0.1
expo-local-authentication@5.0.1
expo-localization@5.0.1
expo-location@5.0.1
expo-mail-composer@5.0.1
expo-media-library@5.0.1
expo-payments-stripe@5.0.1
expo-permissions@5.0.1
expo-print@5.0.1
expo-random@5.0.1
expo-secure-store@5.0.1
expo-sensors@5.0.1
expo-sharing@5.0.1
expo-sms@5.0.1
expo-speech@5.0.2
expo-sqlite@5.0.1
expo-task-manager@5.0.1
expo-video-thumbnails@1.0.1
expo-web-browser@5.0.2
expo-asset@5.0.1
expo-module-template@5.0.2
expo@33.0.3
jest-expo@33.0.2

ide committed:
Make sure expo-ads-facebook builds (TS)

EvanBacon committed:
Moved options button to new file (#4433)

* fixed header tint color for other user profile back button

EvanBacon committed:
[home] Make home more platform agnostic (#4434)

* Delete AddProjectButton.android.js

* Made CloseButton agnostic

* Made Form agnostic

* Made OpenProjectByURLButton agnostic

* Made SearchBar agnostic

* Added platform agnostic QRCodeScreen shim

ide committed:
[mailcomposer] Temporarily ts-ignore expo-mailcomposer

This needs to be fixed, but expo depends on 5.0.1 of mailcomposer so we need to republish 5.0.0 as 5.0.1.

ide committed:
[image-manipulator] Assert that the output of `manipulateAsync` is defined.

We know that `actions.length > 0` so `output` will be defined and can safely add `!`

EvanBacon committed:
Created web version of ExponentKernel (#4435)

ide commented:
Merging this directly will break things, I think. It might be useful as a reference to manually cherry-pick commits from the release branch back to master.

ide committed:
[auth][ios] Backport fix for AppAuth (use input.refreshToken) to SDK 32

Derek pointed this out -- it's correct in 33 but we can fix it in 32 as well.

tsapeta committed:
[ios] bump version to 2.11.1

sjchmiela committed:
[expo] Upgrade react-native-unimodules to 0.4.0 in expo/expo repo

sjchmiela committed:
[tools] Update android-tasks to react-native-unimodules@~0.4.0

sjchmiela committed:
[expokit] Rebuild expokit with updated JSC

sjchmiela committed:
[expokit] Publish expokit@33.0.2 with AppAuth, Android 8.0 fixes and updated JSC

brentvatne committed:
[home] Only show clipboard info on iOS simulator

brentvatne committed:
[web-browser] Bail out from dismissBrowser on Android

CarlAmbroselli committed:
[ios] Add support for iPad slide over and split view (#4466)

sjchmiela committed:
[changelog] Add changelog entry

tsapeta committed:
[expotools] throw nix out of expotools 🎉 (#4469)

# Why

Preparing the ground to #4428 
As far as I know there is no need to utilize nix packages in expotools, however this was always a pain for us since this takes so much time before we can run any command. To be honest I usually run expotools directly from `tools/expotools/bin/expotools.js` to omit nix.

# How

- Removed nix from expotools stub - can we open a champagne already? 🍾 
- Added a wrapper around expotools that ensures build and node_modules folders are up to date
- Made asorted improvements in the output of expotools commands

Screenshot:
<img width="1178" alt="Screenshot 2019-06-07 at 00 22 13" src="https://user-images.githubusercontent.com/1714764/59070333-abda0f00-88ba-11e9-9d9d-a5fdbbec0315.png">

# Test Plan

Tried some different commands with different flags to cover all use cases.

tsapeta committed:
[ios][tools] fix MessageHandlerName in versioned react-native-webview (#4475)

# Why

Fixes #4463 

# How

- Fixed `MessageHandlerName` value in react-native-webview versioned for SDK33.
- Added post-transform rule to versioning script so this hopefully won't happen in the future.
- Improved NCL example to cover postMessage use case.

# Test Plan

`window.ReactNativeWebView.postMessage` works as expected in this new NCL example.

DevJS committed:
[expo] Fix Google Authentication in standalone apps (#4479)

For some reason the previous implementation was performing this check:

```
const iosClientId = Constants.appOwnership === 'standalone' ? config.iosClientId: config.iosStandaloneAppClientId;
```

which was causing issues on a standalone app due to the incorrect clientId being used.

Swapping the keys before making the change fixed the issue. Changing the code to swap the condition will fix the issue for everyone.

mczernek committed:
Correct recent in development API query.

brentvatne committed:
Keep UnavailabilityError when calling dismissBrowser, explicitly check for it in auth session polyfill

brentvatne committed:
[expo] Export a limited set of globals based on usage (#4470)

brentvatne committed:
Polyfill openAuthSessionAsync to wait for browser to close on Android

ide committed:
[web-browser] Bump to 5.0.3 with auth session fixes

brentvatne committed:
[expo] Release 33.0.4

sjchmiela committed:
[android] Remove includeUnimodulesProjects from shell app (#4478)

x-ref: https://github.com/expo/expo-cli/pull/677

This will let us remove `includeUnimodulesProjects` from shell apps so Gradle doesn't sync them upon build

wkozyra95 committed:
[unimodules] ignore MainApplication.java from enabled-modulues/**

tsapeta committed:
[ios] run pod install

tsapeta committed:
[jest] Fix expo-file-system mocks by unscoping methods (#4520)

# Why

Fixes #4503 

# How

Updated `expo-file-system` mock in `jest-expo`

# Test Plan

It must work 😅

sjchmiela committed:
[yarn] Update yarn.lock

sjchmiela committed:
[android] Do not include expo-face-detector and expo-payments-stripe in ejected and standalone apps

brentvatne committed:
[expo] Add SQLite to the Expo global

It is used by typedorm and it's hard to say how many dependents that has, so
this is a safe choice for now until we can send a PR with a clearly better
solution.

brentvatne committed:
[expo] Release 33.0.5

tsapeta committed:
[payments-stripe] fix unimodule.json was missing in published package

ide committed:
[fastlane][ios] Remove background audio for the App Store release

EvanBacon committed:
Optimized expo home assets

EvanBacon committed:
[home] Added image banner for Expo home. (#4424)

* Shimmed out NativeModules.ExponentKernel when running home in home

* shim session storage

* Added banner photo

* Update Profile.js

* Added parallax header

* Updated imports

* Added default banner

* Removed chosen log

* Delete ExponentKernel.js

* Update SearchBar.js

* Update SearchResults.js

* Removed MonoText

* Delete MediaDetailsScreen.tsx

* Update MediaLibraryScreens.ts

* Update MediaLibraryScreen.tsx

* Fixed android bugs

* Update banner-image.png

* Fixed tests

* Update Store.js

* Update ProfileReducer.js

* Update MediaAlbumsScreen.tsx

EvanBacon committed:
Removed the albums screen

ide committed:
[fastlane][ios] Add Expo team information to the Fastfile

tsapeta committed:
[tools] bump @expo/xdl to 54.1.5

ide committed:
[home] Publish new dev home + update embedded bundle

dsokal committed:
[android-shell-app] don't include tools-public/node_modules in the tarball (#4541)

tsapeta committed:
[tools][exponent-view-template] fix iOS shell app failing due to rn-unimodules

ide committed:
[ios] Update camera permission string so it's more relevant in Home

ninjaachibi committed:
[Notification] Fix `scheduleLocalNotification` in android (#4525)

* [notification] add support for sdk-32 scheduling local notifs

* remove Log import

ninjaachibi committed:
[android] bump versions to 2.11.3

brentvatne committed:
[home] Do not render anything if clipboard is invalid

jaulz committed:
[document-picker] Expose DocumentPickerOptions type (#4543)

* fix: export GetDocumentOptions type

* fix: rename GetDocumentOptions to DocumentPickerOptions

bbarthec committed:
[document-picker] Update package

expo-document-picker@5.0.2

tsapeta committed:
[expotools] make expotools work even without installed node modules (#4592)

# Why

`expotools` were not working without an initial `yarn` or `npm install`.

# How

`expotools` wrapper no longer requires any node_modules before it actually runs `yarn install` command. 

# Test Plan

`cd tools/expotools`
`rm -rf node_modules`
`et --help`

brentvatne committed:
[expo] Update @expo/vector-icons in bundledNativeModules.json

brentvatne committed:
[expo] Release 33.0.6

sjchmiela committed:
[android] Add missing native modules to HomeActivity to enable new Home functionalities

sjchmiela committed:
[android] Upgrade lottie-android native dependency to 2.5.6

- fixes crashes on Android 9.0 (https://github.com/expo/expo/issues/4510)
- updates version to the one required by `lottie-react-native`

sjchmiela committed:
[android] Bump Android versions to 2.11.4 (114)

sjchmiela committed:
[expokit] Rebuild expokit with updated Lottie Android

sjchmiela committed:
[expokit] Publish expokit@33.0.3 with latest fixes

dereknelson committed:
muted lottie warning (#4638)

brentvatne committed:
[expo] Fix bundledNativeModules react-native-reanimated version

brentvatne committed:
[expo] Release 33.0.7

esamelson committed:
[android] attempt to open the correct RNBranch library in BranchManager

esamelson committed:
[expokit] gulp update-exponent-view

esamelson committed:
[expokit] publish 33.0.4

wkozyra95 committed:
[android] enable 64-bit in standalone builds for SDK33 (#4565)

* [android] enable 64-bit in standalone builds

* [tools] update @expo/xdl 55.0.14

* [expokit] gulp update-exponent-view
